tests/network/localnet/conftest.py

```diff
@@ -60,60 +69,189 @@ def cudn_localnet(
 ) -> Generator[tuple[BaseVirtualMachine, BaseVirtualMachine]]:
-    vms = (vm_localnet_1, vm_localnet_2)
-    for vm in vms:
-        vm.start()
-    for vm in vms:
-        vm.wait_for_ready_status(status=True)
-        vm.vmi.wait_for_condition(condition=vm.Condition.Type.AGENT_CONNECTED, status=vm.Condition.Status.TRUE)
-    yield vms
+    server_vm, client_vm = run_vms(vms=(vm_localnet_1, vm_localnet_2))
+    yield (server_vm, client_vm)
+
+
+@pytest.fixture()
+def localnet_server(server_client_localnet_vms: tuple[BaseVirtualMachine, BaseVirtualMachine]) -> Generator[Server]:
+    with create_traffic_server(vm=server_client_localnet_vms[0]) as server:
+        assert server.is_running()
+        yield server
+
+
+@pytest.fixture()
+def localnet_client(server_client_localnet_vms: tuple[BaseVirtualMachine, BaseVirtualMachine]) -> Generator[Client]:
+    with create_traffic_client(
+        server_vm=server_client_localnet_vms[0],
+        client_vm=server_client_localnet_vms[1],
+        network_name=LOCALNET_BR_EX_NETWORK,
+    ) as client:
+        assert client.is_running()
+        yield client
+
+
+@pytest.fixture(scope="module")
+def nncp_localnet_on_secondary_node_nic(
+    worker_node1: Node, nodes_available_nics: dict[str, list[str]]
+) -> Generator[libnncp.NodeNetworkConfigurationPolicy]:
+    bridge_name = "localnet-ovs-br"
+    desired_state = libnncp.DesiredState(
+        interfaces=[
+            libnncp.Interface(
+                name=bridge_name,
+                type=NNCP_INTERFACE_TYPE_OVS_BRIDGE,
+                ipv4=libnncp.IPv4(enabled=False),
+                ipv6=libnncp.IPv6(enabled=False),
+                state=libnncp.Resource.Interface.State.UP,
+                bridge=libnncp.Bridge(
+                    options=libnncp.BridgeOptions(libnncp.STP(enabled=False)),
+                    port=[
+                        libnncp.Port(
+                            name=nodes_available_nics[worker_node1.name][-1],
+                        )
+                    ],
+                ),
+            )
+        ],
+        ovn=libnncp.OVN([
+            libnncp.BridgeMappings(
+                localnet=LOCALNET_OVS_BRIDGE_NETWORK,
+                bridge=bridge_name,
+                state=libnncp.BridgeMappings.State.PRESENT.value,
+            )
+        ]),
+    )
+    with libnncp.NodeNetworkConfigurationPolicy(
+        name=bridge_name,
+        desired_state=desired_state,
+        node_selector={WORKER_NODE_LABEL_KEY: ""},
+    ) as nncp:
+        nncp.wait_for_status_success()
+        yield nncp
+
+
+@pytest.fixture(scope="module")
+def cudn_localnet_ovs_bridge(
+    vlan_id: int,
+    namespace_localnet_1: Namespace,
+) -> Generator[libcudn.ClusterUserDefinedNetwork]:
+    with localnet_cudn(
+        name=f"{LOCALNET_OVS_BRIDGE_NETWORK}-cudn",
+        match_labels=LOCALNET_TEST_LABEL,
+        vlan_id=vlan_id,
+        physical_network_name=LOCALNET_OVS_BRIDGE_NETWORK,
+    ) as cudn:
+        yield cudn
```

Comment:
Existing cUDN is waiting for the network to be deployed.
I.e. `cudn.wait_for_status_success()` is missing here.